internal/app/machined/pkg/controllers/k8s/kubelet_static_pod_controller.go

```diff
@@ -317,13 +317,13 @@ func (ctrl *KubeletStaticPodController) refreshPodStatus(ctx context.Context, r
 	podsSeen := map[string]struct{}{}
 
 	for _, pod := range podList.Items {
-		if pod.OwnerReferences != nil {
+		if pod.Metadata.Annotations.ConfigSource != "file" {
```

Comment:
do we need to check if annotations is `nil` (I'm not sure if in some weird case it's not populated?)

Reply:
well, it seems it'll at least have:
```
        "annotations": {
          "kubernetes.io/config.seen": "2022-06-27T14:22:09.108356948Z",
          "kubernetes.io/config.source": "api"
        },
```

but we can check just to be on the safe side.